components/LogWorkout/LogWorkout.jsx

```diff
@@ -146,7 +199,19 @@ const LogWorkout = (props) => {
       <TitleText>{name}</TitleText>
       <StyledFinishButton onPress={() => {
         // TODO: Update Redux and database
-        navigation.goBack();
+        const completedList = [];
+        exerciseState.forEach((exercise) => {
+          exercise.sets.forEach((set) => {
+            completedList.push(set.completed);
+          });
+        });
+        if (completedList.includes(false)) {
```

Comment:
I don't think this is the best way to do this. I think we could replace this with nested for loops that exit on the first false they find. That way if the first one is false we don't waste time on the rest of them and we don't need to call another .includes after to check

Reply:
Also can you take out the TODO now that you've implemented that functionality

Reply:
I get what u mean and that was kinda my first way of doing it so I had like an if false then show alert and return, but I was confused on the logic cuz I didnt know how to do it so that the finish functions only get called if all are completed

Reply:
There's a number of ways you can do it. Here's one off the top of my head

```
let completed = true;
for(let i = 0; completed && i < exerciseState.length; i++) {
   const sets = exerciseState[i].exercise.sets;
   for(let j = 0; completed && j < sets.length; j++) {
      if(!set[j].completed) {
         completed = false;
      }
   }
}

if(!completed) {
   <Show alert>
} else {
   <Do database stuff>
}

```

It might not work exactly correctly but the logic is there

Reply:
wow I rlly forgot about flags that was so incredibly simple my b lmao